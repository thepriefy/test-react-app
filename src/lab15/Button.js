import React from 'react'
import styled from 'styled-components'

export const Button = styled.button`
    color: ${props => props.primary ? 'blue' : 'black' };
`

// class Button extends React.Component {
//     render() {
//         return (
//             <div>
//                 <ButtonCom>click</ButtonCom>
//             </div>
//         )
//     }
// }
//
// export default Button