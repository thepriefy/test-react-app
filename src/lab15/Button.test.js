import React from 'react'
import {shallow} from 'enzyme'
import {Button} from './Button'

describe('Test styled-component', () => {
    it('Should have color blue when primary prop exist', () => {
        const wrapper = shallow(<Button primary />)

        // expect(wrapper).toMatchSnapshot()
        expect(wrapper).toHaveStyleRule('color', 'blue');
    })
})