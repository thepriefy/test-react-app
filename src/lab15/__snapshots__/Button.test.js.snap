// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test styled-component Should have color blue when primary prop exist 1`] = `
.c0 {
  color: blue;
}

<button
  className="c0"
/>
`;
